### PR TITLE
Fixed the mgpu simulator workspace change error

### PR DESCRIPTION
**Pull request by yhwen**
Fixes # .

### Description

Fixed the mgpu simulator meta.json missing issue, caused by workspace change.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/apis/workspace.py [hunk 1]
```diff
@@ -164,8 +164,8 @@ def get_job_meta_path(self, job_id: str) -> str:
     def get_site_privacy_file_path(self):
         return self.get_file_path_in_site_config(WorkspaceConstants.PRIVACY_CONFIG)
 
-    def get_client_custom_dir(self) -> str:
-        return os.path.join(self.get_site_config_dir(), WorkspaceConstants.CUSTOM_FOLDER_NAME)
+    def get_client_custom_dir(self, job_id: str) -> str:
```
Comment (yanchengnv) [inline nvflare/apis/workspace.py:167]: No, this is incorrect!
get_client_custom_dir returns the path to the "custom" folder in "local".
To get the path to job's custom folder, you use get_app_custom_dir.

Please read the comment at the beginning of the workspace.py for the structure of the workspace.

File: nvflare/fuel/hci/client/config.py [hunk 2]
```diff
@@ -39,10 +37,6 @@ def __init__(self, workspace: Workspace):
         base_pkgs = FL_PACKAGES
         module_names = FL_MODULES
 
-        custom_dir = workspace.get_client_custom_dir()
```
Comment (yanchengnv) [inline nvflare/fuel/hci/client/config.py:42]: Why this change? This is to add the client's local/custom to the sys path.

File: nvflare/private/fed/app/client/sub_worker_process.py [hunk 3]
```diff
@@ -310,7 +310,7 @@ def stop(self):
 
 def main(args):
     workspace = Workspace(args.workspace, args.client_name)
-    app_custom_folder = workspace.get_client_custom_dir()
+    app_custom_folder = workspace.get_client_custom_dir(args.job_id)
```
Comment (yanchengnv) [inline nvflare/private/fed/app/client/sub_worker_process.py:313]: Incorrect! We need to add the custom dir of both "local" to sys path.
You should add another line to add the job's custom dir to the sys path.

File: nvflare/private/fed/utils/fed_utils.py [hunk 4]
```diff
@@ -210,7 +210,7 @@ def fobs_initialize(workspace: Workspace = None, job_id: str = None):
 
 def custom_fobs_initialize(workspace: Workspace = None, job_id: str = None):
     if workspace:
-        site_custom_dir = workspace.get_client_custom_dir()
+        site_custom_dir = workspace.get_client_custom_dir(job_id)
```
Comment (yanchengnv) [inline nvflare/private/fed/utils/fed_utils.py:213]: No, this code was meant to register decomposers from the site's local/custom dir.
If you need to register decomposers from the job's custom dir, you need to add code to register workspace.get_app_custom_dir().

**Review by yanchengnv — CHANGES_REQUESTED** [review]
The changes are incorrect. Please see my comments for detail.

File: nvflare/apis/workspace.py (hunk 1 above)
Comment (yhwen) [reply nvflare/apis/workspace.py]: I missed up the get_client_custom_dir() and get_app_custom_dir(). Changed back.

File: nvflare/private/fed/app/client/sub_worker_process.py (hunk 3 above)
Comment (yhwen) [reply nvflare/private/fed/app/client/sub_worker_process.py]: Changed the app_custom_folder to use get_app_custom_dir(args.job_id).

File: nvflare/fuel/hci/client/config.py (hunk 2 above)
Comment (yhwen) [reply nvflare/fuel/hci/client/config.py]: Added back the custom code folder for admin tool.

**Review by yanchengnv — APPROVED** [review]
LGTM
